src/agent/sqlite_manager/include/sqlite_manager.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <SQLiteCpp/SQLiteCpp.h>
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+namespace sqlite_manager
+{
+    const std::map<int, std::string> mapColTypeString {
+        {SQLite::INTEGER, "INTEGER"},
+        {SQLite::TEXT, "TEXT"},
+        {SQLite::FLOAT, "REAL"},
+        {SQLite::BLOB, "BLOB"},
+        {SQLite::Null, "Null"},
+    };
+
+    class Col
+    {
+    public:
+        Col(const std::string& name,
+            const int type,
+            const bool notNull,
+            const bool autoIncr,
+            const bool primary = false)
+            : m_name(name)
+            , m_type(type)
+            , m_notNull(notNull)
+            , m_autoIncrement(autoIncr)
+            , m_primaryKey(primary)
+        {
+        }
+
+        Col(const std::string& name, const int type, const std::string& value)
+            : m_name(name)
+            , m_type(type)
+            , m_value(value)
+        {
+        }
+
+        std::string m_name;
+        int m_type;
+        bool m_notNull;
+        bool m_autoIncrement;
+        bool m_primaryKey;
+        std::string m_value;
+    };
+
+    using Row = std::vector<Col>;
+
+    class SQLiteManager
+    {
+    public:
+        SQLiteManager(const std::string& dbName);
+        ~SQLiteManager() {}
+
+        void CreateTable(const std::string& tableName, const std::vector<Col>& cols);
+        void Insert(const std::string& tableName, const std::vector<Col>& cols);
+        int GetCount(const std::string& tableName);
+        std::vector<Row>
+        Select(const std::string& tableName, const std::vector<Col>& fields, const std::vector<Col>& selCriteria = {});
+        void Remove(const std::string& tableName, const std::vector<Col>& selCriteria = {});
+        void
+        Update(const std::string& tableName, const std::vector<Col>& fields, const std::vector<Col>& selCriteria = {});
+
+    private:
+        /**
+         * @brief Mutex to ensure thread-safe operations.
+         */
+        std::mutex m_mutex;
+
+        /**
+         * @brief The name of the SQLite database file.
+         */
+        const std::string m_dbName;
+
+        /**
+         * @brief Pointer to the SQLite database connection.
+         */
+        std::unique_ptr<SQLite::Database> m_db;
+
+        void ExecuteNoSelectSQL(const std::string& queryString);
```

Comment:
Is this necessary?

Reply:
It will eventually go away, I'm keeping it just in case it comes in handy.